src/battle_ai_switch_items.c

```diff
@@ -1691,6 +1736,25 @@ static s32 GetMaxDamagePlayerCouldDealToSwitchin(u32 battler, u32 opposingBattle
     return maxDamageTaken;
 }
 
+static bool32 CanAbilityTrapOpponent(u16 ability, u32 opponent)
+{
+    if ((B_GHOSTS_ESCAPE >= GEN_6 && IS_BATTLER_OF_TYPE(opponent, TYPE_GHOST)))
+        return FALSE;
+    else if (ability == ABILITY_SHADOW_TAG)
+    {
+        if (B_SHADOW_TAG_ESCAPE >= GEN_4 && GetBattlerAbility(opponent) == ABILITY_SHADOW_TAG) // Check if ability exists in species
```

Comment:
Why does the ability need to be checked again? If you pass the ability into this function isn't it proof that the mon has it?

Reply:
I might be misunderstanding, abilities aren't checked before being passed to `CanAbilityTrapOpponent`, and Shadow Tag specifically needs to be checked on both the AI and the player to see if they cancel each other out.

I think I'm missing what you're talking about specifically, would you mind clarifying? :)

Reply:
oh nevermind. I've only now realized that it is checking opponent. I forgot this ability effect exists.

Reply:
No worries! :D